plugins/orchestrator-backend/src/service/OrchestratorService.test.ts

```diff
@@ -178,26 +197,30 @@ describe('OrchestratorService', () => {
       expect(result).toBe(instances.length);
       expect(dataIndexServiceMock.fetchInstancesTotalCount).toHaveBeenCalled();
     });
-
-    it('should not execute the operation when the cache is empty', async () => {
-      workflowCacheServiceMock.isEmpty = jest.fn().mockReturnValue(true);
-
-      const result = await orchestratorService.fetchInstancesTotalCount();
-
-      expect(result).toBe(0);
-      expect(
-        dataIndexServiceMock.fetchInstancesTotalCount,
-      ).not.toHaveBeenCalled();
-    });
   });
 
   describe('fetchWorkflowOverviews', () => {
     beforeEach(() => {
       jest.clearAllMocks();
     });
 
-    it('should execute the operation when the cache is not empty', async () => {
-      workflowCacheServiceMock.isEmpty = jest.fn().mockReturnValue(false);
+    it('should throw error when data index returns error', async () => {
+      let err: Error | null = null;
+      const errMsg = 'Failed to get workflows overview';
+      sonataFlowServiceMock.fetchWorkflowOverviews = jest
+        .fn()
+        .mockImplementation(() => {
+          throw new Error(errMsg);
+        });
+      try {
+        await orchestratorService.fetchWorkflowOverviews({});
+      } catch (err_) {
+        err = err_ as Error;
+      }
+      expect(err?.message).toEqual(errMsg);
```

Comment:
Is it possible to simplify this into  `expect(promise).rejects.toEqual(errMsg)`?
Eventually, this applies to other sections in this PR.

Reply:
https://github.com/janus-idp/backstage-plugins/pull/1804/commits/6645148eda0872f0c26635e92aace8936fd0f846